### PR TITLE
5 event signup component part 1

### LINKED ISSUE #5: Event Signup Component: Part 1
# Layout: 
Create a new HTML file for the Event Signup. Use existing shared components like the header, footer, and navigation. The main content area should include a form for signing up to events.
# Form: 
The form for the Event Signup should include the following fields:
- Event Name: The name of the community event PiXELL River Financial is signing up for.
- Company Representative's Name: The name of the person who will be representing PiXELL River Financial at the event.
- Representative's Email: The email address of the company representative.
- Role Selection: A dropdown field for selecting the company's intended role at the event (e.g., sponsor, participant, organizer). This will help in planning and managing the company's involvement more effectively.
# Functionality: 
Write a JavaScript function that is triggered when the form is submitted. The function should collect form data and store it in a temporary data object. The function should also validate the form inputs for correctness and completeness, ensuring that all fields have been filled out and the email format is valid.

Jest Test:
- Test that the function is triggered on form submission.
- Test that the function correctly collects form data.
- Test for input validation:
  - Required Fields: Test if the function correctly identifies and flags when any of the required fields (event name, participant's name, participant's email, role) are left empty.
  - Email Validation: Test if the function correctly identifies and flags when the participant's email is not in a valid format.
- Test that the temporary data object is correctly populated with form data.
# Pull Request: 
Create a Pull Request (PR) on GitHub with your changes. Ask your teammates to review and approve your changes. Your project's setup should automatically run all Jest tests when you create the PR. Make sure all tests pass successfully.

### PR DESCRIPTION
**Pull request by awilliamsRRC**
I added the initial form structure and my script so far contains validation. I will be doing jest this weekend.

### CODE REVIEW

File: template.html [hunk 1]
```diff
@@ -18,5 +18,6 @@
     <section id="content"></section>
 
     <footer id="footer">Footer</footer>
+    <script src="event-signup.js"></script>
```
Comment (cbugtai) [inline template.html:21]: template.html is not going to be used its just used as a template for our own htmls.

File: Event Signup/EventSignup.html [hunk 2]
```diff
@@ -4,8 +4,46 @@
...
 </head>
 <body>
-    
+    <!--Header Form template.-->
+    <header>
+        <!--Header Content Here.-->
+    </header>
+    <!--Navigation Form Template-->
+    <nav>
+        <!--Navigation Content Here.-->
```
Comment (cbugtai) [inline Event Signup/EventSignup.html:16]: Use template.html to fill in the header, nav and footer for your own html file

File: Event Signup/event-signup.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    event.preventDefault();
+    // Collect form data.
+    const eventName = document.getElementById('event-name').value;
+    const repName = document.getElementById('rep-name').value;
+    const repEMail= document.getElementById('rep-email').value;
+    const role = document.getElementById('role').value;
+
+    //Validate inputs
+    if(!eventName || !repName || !repEMail || !role){
+        alert('Pleae fill out all required fields.')
```
Comment (cbugtai) [inline Event Signup/event-signup.js]: does this still work if the input is "____"?

Comment (cbugtai) [reply Event Signup/event-signup.js]: otherwise, this is a nice and concise way of determining if an input is empty

**Review by cbugtai — COMMENTED** [review]
No comment on the .js seeing as youre not done there yet but for the html, you could copy the elements from the template, the header, nav and footer and paste it to your html

File: Event Signup/EventSignup.html (hunk 2 above)
Comment (eeluwole1) [reply Event Signup/EventSignup.html:16]: Beautiful

**Review by cbugtai — APPROVED** [review]
Looks like there's an issue with your test but the code looks good!

**Pull request merged**